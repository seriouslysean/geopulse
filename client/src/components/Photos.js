import React from "react";
import Loading from "../components/Loading";

class Photos extends React.Component {
  renderPhotos = () => {
    const collection = this.props.photos.collection;
    return collection.map(photo => {
      const {
        id,
        caption: { text: alt },
        images: { standard_resolution: image },
        likes,
        link,
        user: { username }
      } = photo;
      return (
        <li key={id} className="photos__photo">
          <a rel="external" href={link} title={alt}>
            <div className="photos__image" style={{ backgroundImage: `url(${image.url})` }}>
              <img src={image.url} width={image.width} height={image.height} alt={alt} />
            </div>
            <div className="photos__username">{username}</div>
          </a>
        </li>
      );
    });
  };
  renderContent = () => {
    return (
      <div className="content">
        <ul className="photos__grid">{this.renderPhotos()}</ul>
      </div>
    );
  };
  render() {
    return (
      <section className="section section--spacing section--bg-earth-medium photos">
        <div className="content-container content-container--grid-spacing">
          <div className="photos__content">
            <h1 className="section__title">Photos</h1>

            {this.props.ready ? this.renderContent() : <Loading />}

            <a rel="external" href="https://www.instagram.com/" className="section__attribution" target="_blank">
              Powered by Instagram
            </a>
          </div>
        </div>
      </section>
    );
  }
}

export default Photos;
