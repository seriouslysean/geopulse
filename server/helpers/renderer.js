import React from 'react';
import serialize from 'serialize-javascript';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import { Helmet } from 'react-helmet';
import AppRoutes from '../../client/src/routes/AppRoutes';

export default (req, store, context) => {
  const html = (
    <Provider store={store}>
      <StaticRouter location={req.path} context={context}>
        <div className="app-container">{renderRoutes(AppRoutes)}</div>
      </StaticRouter>
    </Provider>
  );
  const content = renderToString(html);
  const helmet = Helmet.renderStatic();

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />

        ${helmet.title.toString()}
        ${helmet.meta.toString()}

        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/manifest.json">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#43578c">

        <!-- iOS -->
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="apple-mobile-web-app-title" content="Geopulse">
        <link rel="apple-touch-startup-image" href="/assets/geopulse-splash.png">
      
        <!-- Other -->
        <meta name="application-name" content="Geopulse">
        <meta name="theme-color" content="#43578c">
        <meta name="format-detection" content="telephone=no">
      
        <!-- Social -->
        <meta itemprop="name" content="Geopulse">
        <meta itemprop="description" content="Get a pulse on the happenings nearby">
        <meta itemprop="image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
      
        <meta property="og:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        <meta property="og:image:width" content="1200">
        <meta property="og:image:height" content="628">
        <meta property="og:title" content="Geopulse">
        <meta property="og:description" content="Get a pulse on the happenings nearby">
        <meta property="og:url" content="https://geopulse-app.herokuapp.com">
      
        <meta name="twitter:card" content="summary_large_image">
        <meta name="twitter:title" content="Geopulse">
        <meta name="twitter:description" content="Get a pulse on the happenings nearby">
        <meta name="twitter:creator" content="@seriouslysean">
        <meta name="twitter:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        
        <link rel="stylesheet" type="text/css" href="bundle.css" />
      </head>
      <body>
        <div id="root">${content}</div>
        <script>window.INITIAL_STATE = ${serialize(store.getState())}</script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;

  res.send(html);
};
