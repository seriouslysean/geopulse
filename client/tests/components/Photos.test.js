import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Photos from "../../src/components/Photos";
import photosProp from "../../../data/fixtures/photos";

test("should show Photos loading correctly", () => {
  const wrapper = shallow(<Photos />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Photos with content correctly", () => {
  const wrapper = shallow(<Photos ready={true} photos={photosProp} />);
  expect(wrapper).toMatchSnapshot();
});
