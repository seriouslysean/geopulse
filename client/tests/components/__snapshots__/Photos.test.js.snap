// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Photos correctly 1`] = `
<section
  className="section section--spacing section--bg-earth-medium photos"
>
  <div
    className="content-container content-container--grid-spacing"
  >
    <div
      className="photos__content"
    >
      <h1
        className="section__title"
      >
        Photos
      </h1>
      <Loading />
      <a
        className="section__attribution"
        href="https://www.instagram.com/"
        rel="external"
        target="_blank"
      >
        Powered by Instagram
      </a>
    </div>
  </div>
</section>
`;

exports[`should render Photos with content correctly 1`] = `
<section
  className="section section--spacing section--bg-earth-medium photos"
>
  <div
    className="content-container content-container--grid-spacing"
  >
    <div
      className="photos__content"
    >
      <h1
        className="section__title"
      >
        Photos
      </h1>
      <div
        className="content"
      >
        <ul
          className="photos__grid"
        >
          <PhotosPhoto
            key="1667460106995157768_6546293"
            photo={
              Object {
                "attribution": null,
                "caption": Object {
                  "created_time": "1512996756",
                  "from": Object {
                    "full_name": "Sean Kennedy",
                    "id": "6546293",
                    "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                    "username": "seriouslysean",
                  },
                  "id": "17912665465017988",
                  "text": "It’s getting extra Christmasy this morning, hahaha. #🎄 #captaincrunch",
                },
                "comments": Object {
                  "count": 0,
                },
                "created_time": "1512996756",
                "filter": "Clarendon",
                "id": "1667460106995157768_6546293",
                "images": Object {
                  "low_resolution": Object {
                    "height": 320,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/25012870_1945812425670287_2388775529164570624_n.jpg",
                    "width": 320,
                  },
                  "standard_resolution": Object {
                    "height": 640,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/25012870_1945812425670287_2388775529164570624_n.jpg",
                    "width": 640,
                  },
                  "thumbnail": Object {
                    "height": 150,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/25012870_1945812425670287_2388775529164570624_n.jpg",
                    "width": 150,
                  },
                },
                "likes": Object {
                  "count": 15,
                },
                "link": "https://www.instagram.com/p/BckAiH6lMsI/",
                "location": Object {
                  "id": 257206400,
                  "latitude": 39.9231,
                  "longitude": -75.1753,
                  "name": "South Philadelphia",
                },
                "tags": Array [
                  "🎄",
                  "captaincrunch",
                ],
                "type": "image",
                "user": Object {
                  "full_name": "Sean Kennedy",
                  "id": "6546293",
                  "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                  "username": "seriouslysean",
                },
                "user_has_liked": false,
                "users_in_photo": Array [],
              }
            }
          />
          <PhotosPhoto
            key="1630417200979802824_6546293"
            photo={
              Object {
                "attribution": null,
                "caption": Object {
                  "created_time": "1508580897",
                  "from": Object {
                    "full_name": "Sean Kennedy",
                    "id": "6546293",
                    "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                    "username": "seriouslysean",
                  },
                  "id": "17846655562205908",
                  "text": "In case anyone were wondering, he’s been going strong since about 4AM with no sign of stopping. Dude’s a little ball of energy today. 😰👹",
                },
                "comments": Object {
                  "count": 3,
                },
                "created_time": "1508580897",
                "filter": "Lark",
                "id": "1630417200979802824_6546293",
                "images": Object {
                  "low_resolution": Object {
                    "height": 320,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/22709039_1951838335033946_1193752435839991808_n.jpg",
                    "width": 320,
                  },
                  "standard_resolution": Object {
                    "height": 640,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/22709039_1951838335033946_1193752435839991808_n.jpg",
                    "width": 640,
                  },
                  "thumbnail": Object {
                    "height": 150,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/22709039_1951838335033946_1193752435839991808_n.jpg",
                    "width": 150,
                  },
                },
                "likes": Object {
                  "count": 16,
                },
                "link": "https://www.instagram.com/p/BagZ85UFnrI/",
                "location": Object {
                  "id": 257206400,
                  "latitude": 39.9231,
                  "longitude": -75.1753,
                  "name": "South Philadelphia",
                },
                "tags": Array [],
                "type": "image",
                "user": Object {
                  "full_name": "Sean Kennedy",
                  "id": "6546293",
                  "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                  "username": "seriouslysean",
                },
                "user_has_liked": false,
                "users_in_photo": Array [],
              }
            }
          />
          <PhotosPhoto
            key="1619038233836328011_6546293"
            photo={
              Object {
                "attribution": null,
                "caption": Object {
                  "created_time": "1507224419",
                  "from": Object {
                    "full_name": "Sean Kennedy",
                    "id": "6546293",
                    "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                    "username": "seriouslysean",
                  },
                  "id": "17902986109031162",
                  "text": "Oh, hello little grasshopper! #nature",
                },
                "comments": Object {
                  "count": 2,
                },
                "created_time": "1507224419",
                "filter": "Normal",
                "id": "1619038233836328011_6546293",
                "images": Object {
                  "low_resolution": Object {
                    "height": 320,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/22280911_1133560353441845_1541280552114454528_n.jpg",
                    "width": 320,
                  },
                  "standard_resolution": Object {
                    "height": 640,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/22280911_1133560353441845_1541280552114454528_n.jpg",
                    "width": 640,
                  },
                  "thumbnail": Object {
                    "height": 150,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/22280911_1133560353441845_1541280552114454528_n.jpg",
                    "width": 150,
                  },
                },
                "likes": Object {
                  "count": 16,
                },
                "link": "https://www.instagram.com/p/BZ3-rIVlxRL/",
                "location": Object {
                  "id": 257206400,
                  "latitude": 39.9231,
                  "longitude": -75.1753,
                  "name": "South Philadelphia",
                },
                "tags": Array [
                  "nature",
                ],
                "type": "image",
                "user": Object {
                  "full_name": "Sean Kennedy",
                  "id": "6546293",
                  "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                  "username": "seriouslysean",
                },
                "user_has_liked": false,
                "users_in_photo": Array [],
              }
            }
          />
          <PhotosPhoto
            key="1589962572953896235_6546293"
            photo={
              Object {
                "attribution": null,
                "caption": Object {
                  "created_time": "1503758330",
                  "from": Object {
                    "full_name": "Sean Kennedy",
                    "id": "6546293",
                    "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                    "username": "seriouslysean",
                  },
                  "id": "17882358874125646",
                  "text": "Little bean sprout! 🌱",
                },
                "comments": Object {
                  "count": 2,
                },
                "created_time": "1503758330",
                "filter": "Clarendon",
                "id": "1589962572953896235_6546293",
                "images": Object {
                  "low_resolution": Object {
                    "height": 399,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/e35/p320x320/21042322_1517709194944368_5489502188981977088_n.jpg",
                    "width": 320,
                  },
                  "standard_resolution": Object {
                    "height": 799,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/sh0.08/e35/p640x640/21042322_1517709194944368_5489502188981977088_n.jpg",
                    "width": 640,
                  },
                  "thumbnail": Object {
                    "height": 150,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/c0.134.1080.1080/21042322_1517709194944368_5489502188981977088_n.jpg",
                    "width": 150,
                  },
                },
                "likes": Object {
                  "count": 17,
                },
                "link": "https://www.instagram.com/p/BYQrok_FUUr/",
                "location": Object {
                  "id": 257206400,
                  "latitude": 39.9231,
                  "longitude": -75.1753,
                  "name": "South Philadelphia",
                },
                "tags": Array [],
                "type": "image",
                "user": Object {
                  "full_name": "Sean Kennedy",
                  "id": "6546293",
                  "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                  "username": "seriouslysean",
                },
                "user_has_liked": false,
                "users_in_photo": Array [],
              }
            }
          />
          <PhotosPhoto
            key="1579783440601356399_6546293"
            photo={
              Object {
                "attribution": null,
                "caption": Object {
                  "created_time": "1502544883",
                  "from": Object {
                    "full_name": "Sean Kennedy",
                    "id": "6546293",
                    "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                    "username": "seriouslysean",
                  },
                  "id": "17884547467077649",
                  "text": "Mabel has the right idea this morning #💤",
                },
                "comments": Object {
                  "count": 0,
                },
                "created_time": "1502544883",
                "filter": "Normal",
                "id": "1579783440601356399_6546293",
                "images": Object {
                  "low_resolution": Object {
                    "height": 320,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/20686929_1683119255091349_4984418110896340992_n.jpg",
                    "width": 320,
                  },
                  "standard_resolution": Object {
                    "height": 640,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20686929_1683119255091349_4984418110896340992_n.jpg",
                    "width": 640,
                  },
                  "thumbnail": Object {
                    "height": 150,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/20686929_1683119255091349_4984418110896340992_n.jpg",
                    "width": 150,
                  },
                },
                "likes": Object {
                  "count": 17,
                },
                "link": "https://www.instagram.com/p/BXshKtPFphv/",
                "location": Object {
                  "id": 257206400,
                  "latitude": 39.9231,
                  "longitude": -75.1753,
                  "name": "South Philadelphia",
                },
                "tags": Array [
                  "💤",
                ],
                "type": "image",
                "user": Object {
                  "full_name": "Sean Kennedy",
                  "id": "6546293",
                  "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                  "username": "seriouslysean",
                },
                "user_has_liked": false,
                "users_in_photo": Array [],
              }
            }
          />
          <PhotosPhoto
            key="1579058153735451079_6546293"
            photo={
              Object {
                "attribution": null,
                "caption": Object {
                  "created_time": "1502458422",
                  "from": Object {
                    "full_name": "Sean Kennedy",
                    "id": "6546293",
                    "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                    "username": "seriouslysean",
                  },
                  "id": "17892554452014305",
                  "text": "Made a #whole30 breakfast for @imkatiecrafts to start the day off right. YUM. #🥔 #🥑 #🍳",
                },
                "comments": Object {
                  "count": 0,
                },
                "created_time": "1502458422",
                "filter": "Clarendon",
                "id": "1579058153735451079_6546293",
                "images": Object {
                  "low_resolution": Object {
                    "height": 320,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/20686722_674116176120980_5626552005532057600_n.jpg",
                    "width": 320,
                  },
                  "standard_resolution": Object {
                    "height": 640,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20686722_674116176120980_5626552005532057600_n.jpg",
                    "width": 640,
                  },
                  "thumbnail": Object {
                    "height": 150,
                    "url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/20686722_674116176120980_5626552005532057600_n.jpg",
                    "width": 150,
                  },
                },
                "likes": Object {
                  "count": 13,
                },
                "link": "https://www.instagram.com/p/BXp8QZNFfHH/",
                "location": Object {
                  "id": 257206400,
                  "latitude": 39.9231,
                  "longitude": -75.1753,
                  "name": "South Philadelphia",
                },
                "tags": Array [
                  "🥔",
                  "🥑",
                  "whole30",
                  "🍳",
                ],
                "type": "image",
                "user": Object {
                  "full_name": "Sean Kennedy",
                  "id": "6546293",
                  "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12407741_199337137080263_1737628549_a.jpg",
                  "username": "seriouslysean",
                },
                "user_has_liked": false,
                "users_in_photo": Array [
                  Object {
                    "position": Object {
                      "x": 0.8760064166525136,
                      "y": 0.9130434782608695,
                    },
                    "user": Object {
                      "full_name": "Katie Crafts",
                      "id": "971218309",
                      "profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/18949510_830017617161079_6218475533454278656_a.jpg",
                      "username": "imkatiecrafts",
                    },
                  },
                ],
              }
            }
          />
        </ul>
      </div>
      <a
        className="section__attribution"
        href="https://www.instagram.com/"
        rel="external"
        target="_blank"
      >
        Powered by Instagram
      </a>
    </div>
  </div>
</section>
`;
